backend/src/test/java/org/eclipse/tractusx/puris/backend/demandandcapacitynotification/logic/adapter/DemandAndCapacityNotificationSammMapperTest.java

```diff
@@ -0,0 +1,540 @@
+        Partner externalPartner = supplierPartner;
+        Material semiconductorMaterial = new Material(true,
+            false, CUSTOMER_MAT_NUMBER, null, "Semiconductor");
+
+        MaterialPartnerRelation materialPartnerRelation = new MaterialPartnerRelation(semiconductorMaterial,
+            externalPartner,
+            SUPPLIER_MAT_NUMBER,
+            true,
+            false);
+        materialPartnerRelation.setPartnerCXNumber(CX_MAT_NUMBER);
+
+        OwnDemandAndCapacityNotification notification = OwnDemandAndCapacityNotification.builder()
+            .notificationId(UUID.randomUUID())
+            .relatedNotificationId(null)
+            .sourceNotificationId(null)
+            .text("We are in big trouble!")
+            .materials(List.of(semiconductorMaterial))
+            .partner(externalPartner)
+            .effect(EffectEnumeration.DEMAND_INCREASE)
+            .affectedSitesRecipient(externalPartner.getSites().stream().toList())
+            .affectedSitesSender(mySelf.getSites().stream().toList())
+            .startDateOfEffect(dateFromString("25-08-2024 16:00:00"))
+            .expectedEndDateOfEffect(dateFromString("28-08-2024 18:00:00"))
+            .leadingRootCause(LeadingRootCauseEnumeration.PRODUCTION_INCIDENT)
+            .status(StatusEnumeration.OPEN)
+            .build();
+
+        when(mprService.find(semiconductorMaterial, externalPartner)).thenReturn(materialPartnerRelation);
+
+        DemandAndCapacityNotificationSamm samm = demandAndCapacityNotificationSammMapper.ownNotificationToSamm(notification);
+        var jsonString = objectMapper.writeValueAsString(samm);
+        var jsonNode = objectMapper.readTree(jsonString);
+
+        LOG.info(jsonNode::toPrettyString);
+
+        Assertions.assertEquals(jsonNode.get("leadingRootCause").asText(), LeadingRootCauseEnumeration.PRODUCTION_INCIDENT.getValue());
+        Assertions.assertEquals(jsonNode.get("effect").asText(), EffectEnumeration.DEMAND_INCREASE.getValue());
+        Assertions.assertTrue(jsonNode.get("startDateOfEffect").asLong() < jsonNode.get("expectedEndDateOfEffect").asLong());
+        Assertions.assertEquals(jsonNode.get("status").asText(), StatusEnumeration.OPEN.getValue());
+        Assertions.assertTrue(jsonNode.get("affectedSitesSender").toString().contains(mySelf.getSites().getFirst().getBpns()));
+        Assertions.assertTrue(jsonNode.get("affectedSitesRecipient").toString().contains(externalPartner.getSites().getFirst().getBpns()));
+
+        Assertions.assertTrue(jsonNode.get("materialNumberSupplier").isArray());
+        for (var field: jsonNode.get("materialNumberSupplier")) {
+            Assertions.assertEquals(SUPPLIER_MAT_NUMBER, field.asText());
+        }
+
+        Assertions.assertTrue(jsonNode.get("materialNumberCustomer").isArray());
+        for (var field: jsonNode.get("materialNumberCustomer")) {
+            Assertions.assertEquals(CUSTOMER_MAT_NUMBER, field.asText());
+        }
+    }
```

Comment:
Please add check for cx material number as it has been set to `partnerMaterialNumberCx` previously

Reply:
Done